docs/guides/stretch.mdx

```diff
@@ -0,0 +1,197 @@
+    <Tabs>
+    <TabItem value="Invalid" label="Invalid">
+        ```python
+        a = circuit.add_stretch("a")
+        b = circuit.add_stretch("b")
+        circuit.delay(a, (q0, q1))
+        circuit.delay(b, q0)  # Invalid because 2 stretches are applied on q0
+        ```
+     </TabItem>
+     <TabItem value="Valid" label="Valid">
+        ```python
+        a = circuit.add_stretch("a")
+        b = circuit.add_stretch("b")
+        circuit.delay(a, (q0, q1))
+        circuit.delay(b, q2)
+        ```
+     </TabItem>
+    </Tabs>
+
+- The area surrounded by a set of barriers is called a _barrier region_. A stretch variable cannot be used in multiple barrier regions.
+    <Tabs>
+         <TabItem value="Invalid" label="Invalid">
+        ```python
+        # Stretch a is used in two barrier regions
+        a = circuit.add_stretch("a")
+        circuit.barrier((q0, q1))
+        circuit.delay(a, q0)
+        circuit.barrier((q0, q1))
+        circuit.delay(a, q0)
+        circuit.barrier((q0, q1))
+         ```
+
+        ![Illustration of the previous code output](/docs/images/guides/stretch/bad-barrier-region.png 'Invalid use of stretch in barrier regions')
+
+     </TabItem>
+     <TabItem value="Valid" label="Valid">
+        ```python
+        # Stretch a is used inside a barrier region that is on q0 and q1
+        a = circuit.add_stretch("a")
+        circuit.barrier((q0, q1))
+        circuit.delay(a, q0)
+        circuit.barrier(q2)
+        circuit.delay(a, q0)
+        circuit.barrier((q0, q1))
+        ```
+
+        ![Illustration of the previous code output](/docs/images/guides/stretch/good-barrier-region.png 'Valid use of stretch in a barrier region')
+
+     </TabItem>
+    </Tabs>
+
+- Stretch expressions are limited to those of the form `A*stretch + B` where `A` and `B` are floating point or integer constants.
+    <Tabs>
+         <TabItem value="Invalid" label="Invalid">
+        ```python
+        delay[2 * a * a] $0;
+         ```
+     </TabItem>
+     <TabItem value="Valid" label="Valid">
+        ```python
+        qubit[1] q;
+        stretch a;
+        delay[a * 2 + 3dt] q[0];
+        ```
+     </TabItem>
+    </Tabs>
+
+- Stretch expressions can only include a single stretch variable.
+    <Tabs>
+        <TabItem value="Invalid" label="Invalid">
+        ```python
+        qubit[1] q;
+        stretch a;
+        stretch b;
+        delay[a * (b / 1dt)] q[0];
+         ```
+     </TabItem>
+         <TabItem value="Valid" label="Valid">
+        ```python
+        xxxxxx
+        ```
+     </TabItem>
+
+    </Tabs>
+
+- Stretch expressions cannot resolve to negative delay values.
+    <Tabs>
+         <TabItem value="Invalid" label="Invalid">
+        ```python
+        delay[a  - 40.0] $0; #if "a" resolves to a value <= 40.0
+         ```
+     </TabItem>
+     <TabItem value="Valid" label="Valid">
+        ```python
+        xxxxxx
+        ```
+     </TabItem>
+    </Tabs>
```

Comment:
1. `a - 40.0` should probably be `a - 40dt` - you can't add "just a number" to a quantity (`a`) with dimension "time".
2. I'm not clear what this restriction actually is - in the OpenQASM 3 spec, a duration of `a - 40dt` being used is valid and implies that `a >= 40dt`, and the stretch resolver must account for that.  It wouldn't be a solution if `a` resolved to a value `< 40dt` in this situation (note: the comment says `<=` but it probably means `<`; `0dt` presumably _is_ a valid duration).

Reply:
The stretch resolution logic is currently not sophisticated enough to do that. So, if you write:
```
barrier;
stretch a;
delay[a - 40dt] $0;
delay[20dt] $1;
barrier;
```
The solver will not "inflate" the region between the barriers to ensure that `a` is non-negative. Instead, it will just solve `a - 40dt == 20dt` giving `a = -20dt`.

Reply:
If you solve `a - 40dt == 20dt`, wouldn't you get  `a = 60dt` ?  Or does something special happen that I don't understand?

Reply:
Ok ace, thanks - I think an explanation based on your example would make things clearer.

Reply:
Becky, yeah, you're right - I think the example would be `a + 40dt` and `20dt`.